src/components/PopupWithForm.js

```diff
@@ -1,33 +1,41 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-    constructor(popupSelector, handleAddCardFormSubmit) {
+    constructor(popupSelector, handleFormSubmit) {
         super({ popupSelector });
         this._form = this._popup.querySelector('.modal__form');
         this._submitButton = this._popup.querySelector('.modal__button');
         this._formInputs = this._popup.querySelectorAll('input');
-        this._handleAddCardFormSubmit = handleAddCardFormSubmit;
+        this._handleFormSubmit = handleFormSubmit;
+        this._buttonText = this._submitButton.textContent;
     }
 
     _getInputValues() {
         const inputObject = {};
 
         this._formInputs.forEach(input => {
             inputObject[input.name] = input.value;
-        })
+        });
 
         return inputObject;
     }
 
     setEventListeners() {
         super.setEventListeners();
 
-        this._popup.addEventListener('submit', (evt) => {
-            this._handleAddCardFormSubmit(this._getInputValues());
-            evt.preventDefault();
+        this._popup.addEventListener('submit', () => {
+            this._handleFormSubmit(this._getInputValues());
         });
     }
 
+    renderLoading(isLoading) {
+        if(isLoading) {
+            this._submitButton.textContent = "Saving...";
+        } else {
+            this._submitButton.textContent = this._buttonText;
+        }
+    }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`